framework/src/modules/chain/submodules/process_transactions.js

```diff
@@ -276,6 +286,71 @@ class ProcessTransactions {
 		};
 	}
 
+	/**
+	 * Verify user total spending
+	 *
+	 * One user can't spend more than its balance even thought if block contains
+	 * credit transactions settling the balance. In one block total speding must be
+	 * less than the total balance
+	 *
+	 * @param {Array.<Object>} transactions - List of transactions in a block
+	 * @param {StateStore} stateStore - State store instance with prepared account
+	 * @return {Array}
+	 * @private
+	 */
+	// eslint-disable-next-line class-methods-use-this
+	_verifyTotalSpending(transactions, stateStore) {
+		const spendingErrors = [];
+
+		// Group the transactions per senderId to calculate total spending
+		const senderTransactions = transactions.reduce((rv, x) => {
+			(rv[x.senderId] = rv[x.senderId] || []).push(x);
+			return rv;
+		}, {});
+
+		// We need to get the transaction id which cause exceeding the sufficient balance
+		// So we can't sum up all transactions together at once
+		const senderSpending = {};
+		Object.keys(senderTransactions).forEach(senderId => {
```

Comment:
Can we change this to a `for of` giving that it's way faster and we're starting to focus on processing more transactions per block?

Reply:
@pablitovicente We discussed in a meeting previously and we will define this as coding practice , until then use of `forEach` is permissible.